src/content/index.tsx

```diff
@@ -4,20 +4,32 @@ import { GlobalContextProvider } from "./components/GlobalContext/GlobalContext"
 import { Iterations } from "../common/IterationLookup";
 import { QueryManager } from "./lib/utils";
 import { Router } from "./components/Router/Router";
+import { ErrorView } from "./components/ErrorView/ErrorView";
 
 async function main() {
   const [metas, iterationsLookup] = await Promise.all([
-    fetch("/api/metas").then(res => res.json()),
-    fetch("/api/iterations").then(res => res.json()),
+    fetch("/api/metas")
+      .then(res => res.json())
+      .catch(() => {
+        ReactDOM.render(<ErrorView />, document.getElementById("root"));
+      }),
+    fetch("/api/iterations")
+      .then(res => res.json())
+      .catch(() => {
+        ReactDOM.render(<ErrorView />, document.getElementById("root"));
+      }),
   ]);
```

Comment:
Can't you `catch` on the `Promise.all` call?

Reply:
Good idea!